src/datamodel/builder/Payload.cpp

```diff
@@ -10,3 +10,100 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Payload.h"
+namespace uprotocol::datamodel::builder {
+
+// Byte array constructor
+Payload::Payload(const std::vector<uint8_t>& value_bytes,
+                 v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid byte vector payload format");
+	}
+	std::string serialized(value_bytes.begin(), value_bytes.end());
+	payload_ = std::make_tuple(serialized, format);
+}
+
+// String constructor
+Payload::Payload(const std::string& value, v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid String payload format");
+	}
+	payload_ = std::make_tuple(value, format);
+}
+
+// Move string constructor
+Payload::Payload(std::string&& value, v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid RValue String payload format");
+	}
+	payload_ = std::make_tuple(std::move(value), format);
+}
+
+// Move Serialized constructor
+Payload::Payload(Serialized&& serialized) {
+	if (std::get<1>(serialized) <
```

Comment:
Make an enum to replace the magic number for the tuple elements.
Also, surround logical pairs with parenthesis.

Reply:
Looks like the enum comment got lost.